python/lsst/ctrl/pool/pool.py

```diff
@@ -414,6 +415,60 @@ def log(self, source, msg, *args):
             self.out.write("\n")
 
 
+class ReductionThread(threading.Thread):
+    """Thread to do reduction of results
+
+    "A thread?", you say. "What about the python GIL?"
+    Well, because we 'sleep' when there's no immediate response from the
+    slaves, that gives the thread a chance to fire; and threads are easier
+    to manage (e.g., shared memory) than a process.
+    """
+    def __init__(self, reducer, initial=None, sleep=0.1):
+        threading.Thread.__init__(self, name="reducer")
+        self._queue = []  # Queue of stuff to be reduced
+        self._lock = threading.Lock()  # Lock for the queue
+        self._reducer = reducer  # Function that does the reducing
+        self._sleep = sleep  # Time to sleep when there's nothing to do (sec)
+        self._result = initial  # Final result
+        self._done = threading.Event()  # Signal that everything is done
+
+    def _doReduce(self):
+        """Do the actual work
+
+        We pull the data out of the queue and release the lock before
+        operating on it. This stops us from blocking the addition of
+        new data to the queue.
+        """
+        with self._lock:
+            queue = self._queue
+            self._queue = []
+        for data in queue:
+            self._result = self._reducer(self._result, data) if self._result is not None else data
```

Comment:
Serious answer: [RFC-107](https://jira.lsstcorp.org/browse/RFC-107) has not yet been implemented, and KT's directive notes that "Immediate compliance is not required". The [developer guide](https://developer.lsst.io/coding/python_style_guide.html#line-length-must-be-less-than-or-equal-to-110-columns) still gives a limit of 110.  I believe the plan is to automatically reformat every python file before implementing the new width limit.  Given that, and the fact that the rest of the file has a 110 char limit, I believe it's correct to continue to use a 110 char limit until the reformatting has been completed and RFC-107 has been implemented.